transport/tchannel/config.go

```diff
@@ -92,26 +91,39 @@ func (ts *transportSpec) Spec(opts ...Option) config.TransportSpec {
 }
 
 func (ts *transportSpec) buildTransport(tc *TransportConfig, k *config.Kit) (transport.Transport, error) {
-	var config transportConfig
-	// Default configuration.
-	config.tracer = opentracing.GlobalTracer()
-	config.name = k.ServiceName()
-
-	// Apply the spec's transport options.
-	for _, opt := range ts.transportOptions {
-		opt(&config)
+	var cfg transportConfig
+	for _, o := range ts.transportOptions {
+		o(&cfg)
 	}
 
-	// Override options with configuration.
-	if tc.Address != "" {
-		config.addr = tc.Address
+	if cfg.name != "" {
+		return nil, fmt.Errorf("TransportSpec does not accept ServiceName")
 	}
 
-	return config.newTransport(), nil
+	if cfg.addr != "" {
+		return nil, fmt.Errorf("TransportSpec does not accept ListenAddr")
+	}
+
+	if cfg.ch != nil {
+		return nil, fmt.Errorf("TransportSpec does not accept WithChannel")
+	}
+
+	cfg.name = k.ServiceName()
+	return cfg.newTransport(), nil
 }
 
-func (ts *transportSpec) buildInbound(_ *InboundConfig, t transport.Transport, k *config.Kit) (transport.Inbound, error) {
-	return t.(*Transport).NewInbound(), nil
+func (ts *transportSpec) buildInbound(c *InboundConfig, t transport.Transport, k *config.Kit) (transport.Inbound, error) {
+	if c.Address == "" {
+		return nil, fmt.Errorf("inbound address is required")
+	}
+
+	trans := t.(*Transport)
```

Comment:
Should we use comma-ok here and return an error if we get the wrong concrete transport, or is that a critical failure that should panic?